pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -130,10 +130,11 @@ type PodToAdd struct {
 }
 
 // NewDesiredStateOfWorld returns a new instance of DesiredStateOfWorld.
-func NewDesiredStateOfWorld(volumePluginMgr *volume.VolumePluginMgr) DesiredStateOfWorld {
+func NewDesiredStateOfWorld(volumePluginMgr *volume.VolumePluginMgr, operationTimestamps util.OperationStartTimeCache) DesiredStateOfWorld {
 	return &desiredStateOfWorld{
-		nodesManaged:    make(map[k8stypes.NodeName]nodeManaged),
-		volumePluginMgr: volumePluginMgr,
+		nodesManaged:        make(map[k8stypes.NodeName]nodeManaged),
+		volumePluginMgr:     volumePluginMgr,
+		operationTimestamps: operationTimestamps,
```

Comment:
the `operationTimestamps` variable is effectively a package global - will it be cleaner to initialize it as such and have all the code access it directly?

Reply:
Certainly that is an option I have considered. The major thing I was not doing that was the complexity of guarantee key uniqueness across the board with all operations. i.e., provisioning/deletion v.s. attach/detach. It would be a great option if key generation could be managed in a central place as well, however I could not find an easy way or place to put it. WDYT?